package/MDAnalysis/analysis/hbonds/hbond_autocorrel.py

```diff
@@ -427,11 +430,10 @@ def _single_run(self, start, stop):
         box = self.u.dimensions if self.pbc else None
 
         # 2d array of all distances
-        pair, d = capped_distance(self.h.positions, self.a.positions, max_cutoff=self.d_crit, box=box)
-        if self.exclusions:
-            # set to above dist crit to exclude
-            exclude = np.column_stack((self.exclusions[0], self.exclusions[1]))
-            pair = np.delete(pair, np.where(pair==exclude), 0)
+        pair = capped_distance(self.h.positions, self.a.positions, max_cutoff=self.d_crit, box=box,
+                               return_distances=False)
+        if not self.exclusions is None:
+            pair = pair[~ _in2d(pair, self.exclusions)]
```

Comment:
I'm surprised there was no suitable numpy operation for this :/

Reply:
The below passes the analysis suite locally without using a custom Cython function. Maybe double check with Richard if his mixture of Cython/C++ is more efficient -- probably any advantage is related to the integer only nature of the data structure?

```diff
--- a/package/MDAnalysis/analysis/hbonds/hbond_autocorrel.py
+++ b/package/MDAnalysis/analysis/hbonds/hbond_autocorrel.py
@@ -210,6 +210,7 @@ from six import raise_from
 
 import numpy as np
 import scipy.optimize
+from scipy.spatial.distance import cdist
 
 import warnings
 
@@ -433,7 +434,7 @@ class HydrogenBondAutoCorrel(object):
         pair = capped_distance(self.h.positions, self.a.positions, max_cutoff=self.d_crit, box=box,
                                return_distances=False)
         if not self.exclusions is None:
-            pair = pair[~ _in2d(pair, self.exclusions)]
+            pair = pair[~ (cdist(pair, self.exclusions)==0).any(axis=1)]
 
         hidx, aidx = np.transpose(pair)
```

Reply:
You could use cdist but it's odd to use an O(n2) solution when you can just use a set

Reply:
I don't have a strong view on it--if this genuinely a performance-critical part of the code maybe the custom maintenance burden is worth it.